src/sidebar/store/modules/filters.js

```diff
@@ -169,10 +169,21 @@ function changeFocusModeUser(user) {
  * @param {FilterOption} filterOption
  */
 function setFilter(filterName, filterOption) {
-  return {
-    type: actions.SET_FILTER,
-    filterName,
-    filterOption,
+  return (dispatch, getState) => {
+    // If there is a filter conflict with focusFilters, deactivate focus
+    // mode to prevent unintended collisions and let the new filter value
+    // take precedence.
+    if (getState().filters.focusFilters?.[filterName]) {
```

Comment:
For some reason I'm having trouble wrapping my head around what you're envisioning, would you mind elaborating a little bit?

Reply:
If you want one action to incorporate the effects of another, then one option is to use a thunk with `dispatch` as you have done here. Another is to call one reducer from within another:

```diff
diff --git a/src/sidebar/store/modules/filters.js b/src/sidebar/store/modules/filters.js
index e36e7eff5..c47b6f5a8 100644
--- a/src/sidebar/store/modules/filters.js
+++ b/src/sidebar/store/modules/filters.js
@@ -119,6 +119,15 @@ const update = {
   },
 
   SET_FILTER: function (state, action) {
+    // If there is a filter conflict with focusFilters, deactivate focus
+    // mode to prevent unintended collisions and let the new filter value
+    // take precedence.
+    let newState = /** @type {typeof state} */ ({});
+    if (state.focusFilters?.[action.filterName]) {
+      // eslint-disable-next-line new-cap
+      newState = update.SET_FOCUS_MODE(state, { active: false });
+    }
+
     const updatedFilters = {
       ...state.filters,
       [action.filterName]: action.filterOption,
@@ -127,7 +136,9 @@ const update = {
     if (action.filterOption?.value === '') {
       delete updatedFilters[action.filterName];
     }
-    return { filters: updatedFilters };
+    newState.filters = updatedFilters;
+
+    return newState;
   },
 
   SET_FILTER_QUERY: function (state, action) {
@@ -170,22 +181,7 @@ function changeFocusModeUser(user) {
  * @param {FilterOption} filterOption
  */
 function setFilter(filterName, filterOption) {
-  return (dispatch, getState) => {
-    // If there is a filter conflict with focusFilters, deactivate focus
-    // mode to prevent unintended collisions and let the new filter value
-    // take precedence.
-    if (getState().filters.focusFilters?.[filterName]) {
-      dispatch({
-        type: actions.SET_FOCUS_MODE,
-        active: false,
-      });
-    }
-    dispatch({
-      type: actions.SET_FILTER,
-      filterName,
-      filterOption,
-    });
-  };
+  return { type: actions.SET_FILTER, filterName, filterOption };
 }
 
 /** Set the query used to filter displayed annotations. */
```

There isn't a significant advantage to changing the current code to the above in this case. I'm just noting that it would have been an option.

Reply:
Admittedly did not know this was even possible, so this has value just as educational content :).